### PR TITLE
Performance:

### PR DESCRIPTION
**Pull request by arturobernalg**
* Use arrayCopy
* Avoid redundant String.format
* Exit the loop early

### CODE REVIEW

File: src/main/org/apache/tools/ant/taskdefs/compilers/JavacExternal.java [hunk 1]
```diff
@@ -94,9 +94,8 @@ private int moveArgFileEligibleOptionsToEnd(String[] commandLine) {
         for(int i = nonArgFileOptionIdx + 1; i < commandLine.length; i++) {
             if (!isArgFileEligible(commandLine[i])) {
                 String option = commandLine[i];
-                for(int j = i - 1; j >= nonArgFileOptionIdx; j--) {
-                    commandLine[j + 1] = commandLine[j];
-                }
+                if (i - nonArgFileOptionIdx >= 0)
```
Comment (bodewig) [inline src/main/org/apache/tools/ant/taskdefs/compilers/JavacExternal.java:97]: please add braces even for single line blocks

**Comment by bodewig** [comment]
"performance" is such a big word, I very much doubt there will be any measurable effect on a real build by the changes :-)

anyway, thanks for the patch @arturobernalg . Please bring back the braces and please add yourself to the CONTRIBUTORS and contributors.xml files (unless you are already there and I just didn't see it).

File: src/main/org/apache/tools/ant/taskdefs/compilers/JavacExternal.java (hunk 1 above)
Comment (arturobernalg) [reply src/main/org/apache/tools/ant/taskdefs/compilers/JavacExternal.java]: HI @bodewig 
I just add the braces.
TY

**Comment by bodewig** [comment]
Thank you @arturobernalg

**Pull request merged**